prometheus/histogram.go

```diff
@@ -747,9 +749,10 @@ func (h *histogram) Write(out *dto.Metric) error {
 	waitForCooldown(count, coldCounts)
 
 	his := &dto.Histogram{
-		Bucket:      make([]*dto.Bucket, len(h.upperBounds)),
-		SampleCount: proto.Uint64(count),
-		SampleSum:   proto.Float64(math.Float64frombits(atomic.LoadUint64(&coldCounts.sumBits))),
+		Bucket:           make([]*dto.Bucket, len(h.upperBounds)),
+		SampleCount:      proto.Uint64(count),
+		SampleSum:        proto.Float64(math.Float64frombits(atomic.LoadUint64(&coldCounts.sumBits))),
+		CreatedTimestamp: h.createdTs,
```

Comment:
@bwplotka how should this handle native histogram resets?  The correct start time for a fixed-bucket histogram is the instrument creation time, but the correct (I think?) start time for native histograms is the last reset time.

Reply:
I still haven't had the chance to work with Native Histograms... May I ask why do they need to be reset in the first place?

Reply:
Yup, didn't think about it yet. I would say scoping this to exclude native histograms (explicitly) could be  one solution. 

Asking @beorn7 is another solution =D

I am diving into native histograms more next week, so will know more.

Reply:
> reseting CT on native histogram reset sounds trivial

I know I said that during our sync today, but now that I tried to implement it doesn't look so simple 😅. It could be just me not understanding the concepts of native histograms though.

I can reset functions being called in `histogram.Write()`, does that mean that histograms are reset on every scrape? I feel like I'm missing something here 🤔

Reply:
Native histograms are reset if they contain too many buckets _and_ a sufficient time has passed since the last reset. All of this is configurable, see https://pkg.go.dev/github.com/prometheus/client_golang/prometheus#HistogramOpts .

I'm fairly certain that we should reset the CT together with the histogram. After all, one of the cited benefits of the CT is to detect resets reliably.

Reply:
Fixed in PR to this PR https://github.com/ArthurSens/client_golang/pull/1 (: